constraint/pkg/regorewriter/regorewriter_test.go

```diff
@@ -495,3 +506,256 @@ func TestRegoRewriterErrorCases(t *testing.T) {
 		}
 	}
 }
+
+func TestRegoRewriterRewriteDataRef(t *testing.T) {
+	tests := []struct {
+		name    string
+		prefix  string
+		imports string
+		extern  []string
+		want    string
+	}{
+		{
+			name:    "no rewrite when data ref is an external reference",
+			prefix:  "foo.bar",
+			imports: "data.ext",
+			extern:  []string{"data.ext", "data.inventory"},
+			want:    "data.ext",
+		},
+		{
+			name:    "rewrite for ref is sub of external references",
+			prefix:  "foo.bar",
+			imports: "data.ext",
+			extern:  []string{"data.ext.util"},
+			want:    "data.foo.bar.ext",
+		},
+		{
+			name:    "no rewrite for non data ref",
+			prefix:  "foo.bar",
+			imports: "invalid_data.name",
+			want:    "invalid_data.name",
+		},
+		{
+			name:    "add prefix for data ref",
+			prefix:  "foo.bar",
+			imports: "data.some_ref",
+			want:    "data.foo.bar.some_ref",
+		},
+		{
+			name:    "add prefix same as libs",
+			prefix:  "data.lib",
+			imports: "data.some_ref",
+			want:    "data.data.lib.some_ref",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			externs := []string{"data.inventory"}
+			if tt.extern != nil {
+				externs = tt.extern
+			}
+			r, err := New(NewPackagePrefixer(tt.prefix), []string{"data.lib"}, externs)
+			if err != nil {
+				t.Fatalf("Failed to create %s", err)
+			}
+			ref, err := ast.ParseRef(tt.imports)
+			if err != nil {
+				t.Fatalf("Failed to parse ref %s", err)
+			}
+			wantRef, err := ast.ParseRef(tt.want)
+			if err != nil {
+				t.Fatalf("Failed to parse ref %s", err)
+			}
+			if got := r.rewriteDataRef(ref); !got.Equal(wantRef) {
+				t.Errorf("rewriteDataRef() = %q, want %v", got, tt.want)
```

Comment:
Nit: Since `got` and `want` are of the same type, either use `%q` or `%+v` for both of them so they're printed the same way.